src/ngraph/validation_util.cpp

```diff
@@ -922,84 +929,138 @@ void ngraph::opset1::infer_conv_backprop_auto_padding(const Shape& input_data_sh
 
 namespace
 {
-    /// \brief Scalar variant describes value of an Output
+    /// \brief Scalar variant describes value of an Output, for use in max shape determination
+    ///
+    /// For tensor values, we use the maximum value in the tensor
     struct MaxValue
     {
         /// \brief No information known about the output
-        MaxValue()
-            : m_element_type(element::Type_t::undefined)
+        MaxValue() {}
+        /// \brief uint64_t assoiated with the output
+        MaxValue(uint64_t value)
+            : m_value(value)
         {
         }
-        /// \brief int64_t assoiated with the output
-        MaxValue(int64_t value)
-            : m_element_type(element::Type_t::i64)
-            , m_int64_t(value)
+        MaxValue(vector<uint64_t> slices, int64_t slice_axis)
+            : m_slices(slices)
```

Comment:
Either pass vector by const ref or use `m_slices(std::move(slices))` to save a copy. This is doing two copies.

Reply:
Got it.